go/common/amount/amount.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package amount
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/holiman/uint256"
+)
+
+// Amount is a 256-bit unsigned integer used for token values like balances.
+type Amount struct {
+	internal uint256.Int
+}
+
+// NewAmount creates a new U256 Amount from up to 4 uint64 arguments. The
+// arguments are given in the Big Endian order. No argument results in a value of zero.
+// The constructor panics if more than 4 arguments are given.
+func NewAmount(args ...uint64) Amount {
+	if len(args) > 4 {
+		panic("too many arguments")
+	}
+	result := Amount{}
+	offset := 4 - len(args)
+	for i := 0; i < len(args) && i < len(result.internal); i++ {
```

Comment:
```suggestion
	for i := 0; i < len(args); i++ {
```

Reply:
correct me if I am wrong, but I guess this extra check is redundant, right? The input is max 4 items and the internal repres. of uin256.Int is also 4 items

Reply:
Indeed, it might be simplified.